src/examples/java/io/nats/examples/jetstream/ResilientPublisher.java

```diff
@@ -36,10 +39,29 @@
  */
 public class ResilientPublisher implements Runnable {
     public static void main(String[] args) {
-        try (Connection nc = Nats.connect()) {
+        Options options = Options.builder()
+            .socketWriteTimeout(20_000)
+            .connectionListener((conn, type) -> System.out.println(type))
+            .errorListener(new ErrorListenerConsoleImpl())
+            .build();
+        try (Connection nc = Nats.connect(options)) {
+
+// JetStream PUBLISHER EXAMPLE
             JetStreamManagement jsm = nc.jetStreamManagement();
-            createOrReplaceStream(jsm, "stream", StorageType.Memory, "subject");
-            ResilientPublisher rp = newInstanceReportingAndDelay(jsm, "stream", "subject", "prefix", 100, 1000);
+            createOrReplaceStream(jsm, "js-stream", StorageType.Memory, "js-subject");
+            ResilientPublisher rp = new ResilientPublisher(nc, jsm, "js-stream", "js-subject")
+                .basicDataPrefix("data")
+                .delay(1)
+                .reportFrequency(1000);
+// END JetStream PUBLISHER EXAMPLE
+
+// CORE PUBLISHER EXAMPLE
+//            ResilientPublisher rp = new ResilientPublisher(nc, "core-subject")
+//                .basicDataPrefix("data")
+//                .delay(1)
+//                .reportFrequency(1000);
+// END CORE PUBLISHER EXAMPLE
+
```

Comment:
This is an example class that demonstrates resilient publishing. I added some functionality to it, for instance it only demoed Jet Stream publishing, but now can support core publishing. There are also some other improvements.